spotbugs/src/main/java/edu/umd/cs/findbugs/util/MutableClasses.java

```diff
@@ -147,20 +156,55 @@ private boolean looksLikeASetter(Method method) {
             return false;
         }
 
-        // If the method throws an UnsupportedOperationException then we ignore it.
-        ExceptionTable exceptions = method.getExceptionTable();
-        if (exceptions != null) {
-            if (Arrays.asList(exceptions.getExceptionNames()).contains("java.lang.UnsupportedOperationException")) {
-                return false;
+        private String getSig() {
+            String local = sig;
+            if (local == null) {
+                sig = local = "L" + cls.getClassName().replace('.', '/') + ";";
+            }
+            return local;
+        }
+
+        private boolean isImmutableByContract() {
+            Boolean local = immutableByContract;
```

Comment:
I don't see why the local variable is being used when we're populating the instance field anyway.

Reply:
Ditto.